ProcessLib/HydroMechanics/HydroMechanicsFEM.h

```diff
@@ -0,0 +1,425 @@
+            auto const& dNdx_p = _ip_data[ip]._dNdx_p;
+
+            auto const& B = _ip_data[ip]._b_matrices;
+            auto const& sigma = _ip_data[ip]._sigma;
+
+            auto const& C = _ip_data[ip]._C;
+
+            double const S =
+                _process_data.storage_coefficient(t, x_position)[0];
+            double const K_over_mu =
+                _process_data.intrinsic_permeability(t, x_position)[0] /
+                _process_data.fluid_viscosity(t, x_position)[0];
+            auto const alpha = _process_data.biot_coefficient(t, x_position)[0];
+            auto const rho_sr = _process_data.solid_density(t, x_position)[0];
+            auto const rho_fr = _process_data.fluid_density(t, x_position)[0];
+            auto const porosity = _process_data.porosity(t, x_position)[0];
+            auto const body_force =
+                _process_data.specific_body_force(t, x_position);
+            assert(body_force.size() == DisplacementDim);
+            auto const b =
+                Eigen::Map<typename ShapeMatricesType::template VectorType<
+                    DisplacementDim> const>(body_force.data(), DisplacementDim);
+
+            auto const& identity2 = MaterialLib::SolidModels::Invariants<
+                kelvin_vector_size>::identity2;
+
+            //
+            // displacement equation, displacement part
+            //
+            _ip_data[ip].updateConstitutiveRelation(t, x_position, dt, u);
+
+            local_Jac
+                .template block<displacement_size, displacement_size>(
+                    displacement_index, displacement_index)
+                .noalias() += B.transpose() * C * B * w;
+
+            typename ShapeMatricesType::template MatrixType<DisplacementDim,
+                                                            displacement_size>
+                N_u = ShapeMatricesType::template MatrixType<
+                    DisplacementDim,
+                    displacement_size>::Zero(DisplacementDim,
+                                             displacement_size);
+            for (int i = 0; i < DisplacementDim; ++i)
+                N_u.template block<1, displacement_size / DisplacementDim>(
+                       i, i * displacement_size / DisplacementDim)
+                    .noalias() = _ip_data[ip]._N_u;
+
+            double const rho = rho_sr * (1 - porosity) + porosity * rho_fr;
+            local_rhs
+                .template block<displacement_size, 1>(displacement_index, 0)
+                .noalias() -=
+                (B.transpose() * sigma - N_u.transpose() * rho * b) * w;
+
+            //
+            // displacement equation, pressure part
+            //
+            Kup.noalias() += B.transpose() * alpha * identity2 * N_p * w;
+
+            //
+            // pressure equation, pressure part.
+            //
+            laplace_p.noalias() += dNdx_p.transpose() * K_over_mu * dNdx_p * w;
+
+            storage_p.noalias() += N_p.transpose() * S * N_p * w;
```

Comment:
storage_p --> mass_p

Besides, if the general fluid property models are use for fluid viscosity and density, please refer to   LiquidFlowMaterialProperties::getMassCoefficient(...) to get rho and drho_dp. With the changes, S can be replaced with  porosity * drho_dp / rho + S.